nvaccess/nvda
# Log WindowErrors for easeOfAccess and config

### Link to issue number:

Relates to #13155 

### Summary of the issue:

In order to investigate #13155, further logging is needed.

### Description of how this pull request fixes the issue:

Adds logging for WindowErrors raised in easeOfAccess and config.  This includes improvements to functions unrelated to #13155, which access the registry, by adding logging to prevent similar issues.

### Testing strategy:

- [x] General alpha testing is needed to ensure the warnings and errors raised aren't noisy.
- [x] Manually test installing NVDA
- [x] Manually test toggling NVDA start after logon (both on and off)
   - ensure changing the setting persists across instances
     - from the installer
     - [from an installed copy
- [x] Manually test toggling NVDA start on logon screen (both on and off)
   - ensure changing the setting persists across instances
     - from the installer
     - from an installed copy
- [x] Additional testing is needed to try and reproduce #13155, and get a clear picture of where the problem could be occurring.
     - This is yet to be reliably reproduced. The following has been tested:
          - Running the installer under a profile with special characters which can cause os.stat to fail (`Testé ç ş ğ`).
          - [x] #13155 general re-attempt
              - download a copy of the installer from this PR
              - install this PR build
              - turn on "start after logon", save the config
              - create a copy of the installer from this PR, identical except with a higher build number
              - run the identical, but newer installer to cause an update to occur
              - capture the installer log
              - start the newly installed copy
              - open the settings, check if "start after logon" is enabled
              - if not, capture the latest NVDA log
              - upload both logs

### Known issues with pull request:

None

### Change log entries:

For Developers

```t2t
- ```config.RUN_REGKEY``, ``config.NVDA_REGKEY`` are deprecated, please use ``config.RegistryKey.RUN``, ``config.RegistryKey.NVDA`` instead. These will be removed in 2023. (#13242)
- ``easeOfAccess.ROOT_KEY``, ``easeOfAccess.APP_KEY_PATH`` are deprecated, please use``easeOfAccess.RegistryKey.ROOT``, ``easeOfAccess.RegistryKey.APP`` instead. These will be removed in 2023. (#13242)
- ``easeOfAccess.APP_KEY_NAME`` has been deprecated, to be removed in 2023. (#13242)
```

### Code Review Checklist:



  - description is up to date
  - change log entries
  - Unit tests
  - System (end to end) tests
  - Manual testing
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
